packages/gatsby-source-contentful/src/source-nodes.js

```diff
@@ -58,16 +59,22 @@ export async function sourceNodes(
     actions
   const online = await isOnline()
 
-  getNodes().forEach(node => {
-    if (node.internal.owner !== `gatsby-source-contentful`) {
-      return
-    }
-    touchNode(node)
-    if (node?.fields?.localFile) {
-      // Prevent GraphQL type inference from crashing on this property
-      touchNode(getNode(node.fields.localFile))
-    }
-  })
+  // Gatsby only checks if a node has been touched on the first sourcing.
```

Comment:
Just curious, why is this?

Reply:
Mostly that it's an expensive check to figure out if a node should be garbage collected and we want subsequent sourcing to be as fast as possible. The assumption is that if a source plugin doesn't know how to delete nodes properly, you'll just have to restart the build runner now and then

Reply:
Makes sense, thanks!